addons/controllers/clusterbootstrap_handlers.go

```diff
@@ -76,6 +79,52 @@ func (r *ClusterBootstrapReconciler) ClusterBootstrapToClusters(o client.Object)
 	return []ctrl.Request{{NamespacedName: client.ObjectKeyFromObject(cluster)}}
 }
 
+// SecretsToClusters is the Map Function for watching Secrets that filters the events on
+// objects of type secret and returns requests for reconcile if required
+func (r *ClusterBootstrapReconciler) SecretsToClusters(o client.Object) []ctrl.Request {
+	secret, ok := o.(*corev1.Secret)
+	if !ok {
+		r.Log.Error(errors.New("invalid type"),
+			"Expected to receive Secret resource",
+			"actualType", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	log := r.Log.WithValues(constants.SecretNameLogKey, secret.Name)
+	log.V(4).Info("Mapping secrets to cluster")
+
+	// Here we filter based on two criteria
+	// 1. Secrets having the type ClusterBootstrapManagedSecret, OR
+	// 2. Secrets with ClusterNameLabel set
+	// For secrets that are cloned by ClusterBootstrap, we set the type and the first filter is used. We also
+	// set the ClusterNameLabel on these cloned secrets
+	// For other secrets such as those that we get from provider status (extensible provider model), the second
+	// filter is used because it is not possible to patch the Type of these secrets because the Type may be immutable
+	//
+	// Need to confirm: We can just use the second filter that is based on the cluster label for all secrets.
```

Comment:
suggestion: cleanup L104 before merging.
question: I had the same question, filter #2 is a super set of filter #1, correct ?

Reply:
Yes it is